staging/src/k8s.io/component-base/cli/globalflag/globalflags.go

```diff
@@ -19,35 +19,32 @@ package globalflag
 import (
 	"flag"
 	"fmt"
+	"os"
 	"strings"
 
 	"github.com/spf13/pflag"
-
 	"k8s.io/component-base/logs"
+	"k8s.io/klog"
 )
 
 // AddGlobalFlags explicitly registers flags that libraries (klog, verflag, etc.) register
 // against the global flagsets from "flag" and "k8s.io/klog".
 // We do this in order to prevent unwanted flags from leaking into the component's flagset.
 func AddGlobalFlags(fs *pflag.FlagSet, name string) {
-	addGlogFlags(fs)
+	addKlogFlags(fs)
 	logs.AddFlags(fs)
 
 	fs.BoolP("help", "h", false, fmt.Sprintf("help for %s", name))
 }
 
-// addGlogFlags explicitly registers flags that klog libraries(k8s.io/klog) register.
-func addGlogFlags(fs *pflag.FlagSet) {
-	// lookup flags of klog libraries in global flag set and re-register the values with our flagset
-	Register(fs, "logtostderr")
-	Register(fs, "alsologtostderr")
-	Register(fs, "v")
-	Register(fs, "skip_headers")
-	Register(fs, "stderrthreshold")
-	Register(fs, "vmodule")
-	Register(fs, "log_backtrace_at")
-	Register(fs, "log_dir")
-	Register(fs, "log_file")
+// addKlogFlags adds flags from k8s.io/klog
+func addKlogFlags(fs *pflag.FlagSet) {
+	local := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	klog.InitFlags(local)
```

Comment:
This has the side effect of resetting the values in klog to defaults every time a new flagset is constructed.